File: cases/taint/go/element-object-positive/element_object_positive.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: element-object-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: element-object-sink

type Item struct {
	Value string
}

func run() {
	items := []Item{{Value: "clean"}, {Value: "clean"}}
	items[0].Value = dfb_source() // DFB-WITNESS: element-object-store
	dfb_sink(items[0].Value)
}

File: cases/taint/javascript/element-object-positive/ElementObjectPositive.js
function dfb_source() { // DFB-SOURCE: element-object-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: element-object-sink

function run() {
  const items = [{ value: "clean" }, { value: "clean" }];
  items[0].value = dfb_source(); // DFB-WITNESS: element-object-store
  dfb_sink(items[0].value);
}
